GeoJSON revamp: data in repo

brockfanning opened the pull request:
Fixes #109 

This depends on https://github.com/open-sdg/sdg-build/pull/99 so a full review needs to be done of that PR as well as this one.

## Breaking changes

This changes the map functionality so that it expects the GeoJSON to be in the data repository. This needs to happen in the data repository via sdg-build. So, if a country upgrades to this version of open-sdg without also upgrading sdg-build versions, their maps will stop functioning.

Along the same lines, the map configuration in _config.yml needs to change.

## Overview of changes

1. The map no longer looks for a single GeoJSON file at a hardcoded location. Instead, it looks for individual GeoJSON files in the data repo that have been pre-built for each indicator.
2. The GeoJSON download button on the maps has been removed, and standard links have been added below the map. Some benefits of this:
    a. They use the same style as other download buttons in the platform
    b. The download is a simple link, so it can be reached by scripts (interoperability)
    c. Instead of confusingly only downloading the visible layer, it now displays all the layers as separate buttons
3. A lot of Javascript code in open-sdg has been removed because it isn't needed
4. Several javascript libraries are no longer being loaded on non-map pages (saves bandwidth)
5. I've removed any reference to the `data_geocode_regex` metadata field. I think this was non-functional anyway, and we would need to reimplement that functionality if it's still wanted.
6. The GeoJSON files now contains all the disaggregation data, though it's currently just using the first one. (After this is merged I'll post a followup PR that displays all the disaggregation, fixing #374.)
7. The options in the map configuration have changed.

## Screenshots

Here is an example of the new download button (the "My English regions" is configurable and just demonstrates that it is translated):
![Screenshot from 2020-02-11 09-52-01](https://user-images.githubusercontent.com/1319083/74247607-4ee9f080-4cb4-11ea-9d0b-3db69662d884.png)

brockfanning commented:
@LucyGwilliamAdmin @SavvasStephanides A lot to go over with this one. We could discuss in the next call. Testing is a bit of a hassle because it depends on an sdg-build change.

christoph4711 commented:
Where I can test the new map functionalities?
And how can I update my starter repo? (Sorry for this question, but I am not familiar with this github stuff)

brockfanning commented:
@christoph4711 Can you check what version of open-sdg you're using now? It would be in the `remote_theme` line of `_config.yml`. Also can you check the version of jekyll-open-sdg-plugins that you're using (this would be in `Gemfile`). Finally, in the data repository, what version of sdg-build are you using? (this would be in the `scripts/requirements.txt` file in the data repository).

The reason I ask is that in order to test cutting-edge features, you'll need to be using the latest versions of all those things.

Also for your awareness, we are on track to release a stable version 1.0.0 of Open SDG some time in the next month. When we release that, we'll include thorough upgrade instructions. So it may be more efficient to hold off.

In the meantime, I'll try to add some screenshots to this PR so you can see any visual changes.

brockfanning commented:
One nice thing about this - this makes it almost trivial to add a disaggregation drop-down to the maps. In fact, I'm almost tempted to add it to this PR... Here's an example of what it might look like:
![Screenshot from 2020-02-11 09-48-51](https://user-images.githubusercontent.com/1319083/74247395-fc103900-4cb3-11ea-95db-ab9e722d94cc.png)

christoph4711 commented:
@brockfanning 
This feature would be great! I try to hold off  until 1.0.0. :-)

brockfanning commented:
Since it was so easy to add, I went ahead and added the disaggregation select to this PR. I've updated the PR description and title.

brockfanning commented:
@AnnCorp @SavvasStephanides @LucyGwilliamAdmin I've removed the disaggregation select from this PR, and updated the description/title. I can make a follow-up PR after this for continuing with that feature.

Any review/testing is welcome! As a reminder, it depends on another PR in sdg-build.

brockfanning commented:
@SavvasStephanides @LucyGwilliamAdmin This should be a bit easier to test, now that the sdg-build change has been merged.

brockfanning commented:
@SavvasStephanides @LucyGwilliamAdmin Just a note that I resolved the merge conflicts on this one.

LucyGwilliamAdmin commented:
@brockfanning @SavvasStephanides really sorry so I think I've missed something about the make serve stuff?

How do I test this using make serve?

brockfanning commented:
@LucyGwilliamAdmin In theory, if you have the required software installed (Ruby, Python, Node.js) you can test it by pull my branch and running `make serve` in the root of the repository. If you don't already have my remote set up in Git, the command would be:
```
git remote add brock https://github.com/brockfanning/open-sdg.git
```
Then you could do this:
```
git fetch brock
git checkout geojson-in-data-repo
make serve
```

LucyGwilliamAdmin commented:
Thanks @brockfanning 

So I've got all the software and this worked right up until the last line (`make serve`), where I got the error:
'make' is not recognized as an internal or external command

brockfanning commented:
@LucyGwilliamAdmin Ah, I think that it doesn't work in Windows. I guess this is a good reason we need the Docker solution @SavvasStephanides is working on.

LucyGwilliamAdmin commented:
Ok no worries, thanks for you help though! I think I'll have to leave this one all to you @SavvasStephanides

SavvasStephanides commented:
@LucyGwilliamAdmin If you can run Docker you can give it a go https://github.com/open-sdg/open-sdg/issues/197#issuecomment-594548506

SavvasStephanides commented:
@brockfanning Do you have an indicator I can test this on?

brockfanning commented:
@SavvasStephanides The tests for this use indicator 1.3.1.

SavvasStephanides commented:
Thanks @brockfanning ! Seems to be working nicely. I'll have a quick check on the code and approve.

SavvasStephanides commented on _includes/assets/js/plugins/jquery.sdgMap.js:
```diff
@@ -2,7 +2,11 @@
-(function($, L, chroma, window, document, undefined) {
+(function($) {
+
+  if (typeof L === 'undefined') {
```
Where does `L` point to here, if you removed it from the function's arguments?

brockfanning replied:
@SavvasStephanides That's a global variable created by Leaflet. It's not there, I assume that Leaflet did not get successfully loaded.

SavvasStephanides replied:
Ah fair enough. Thanks!
